x-pack/plugins/cases/server/common/models/case_with_comments.ts

```diff
@@ -293,9 +313,31 @@ export class CaseCommentModel {
       references = [...references, ...commentStringReferences];
     }
 
+    if (
+      commentReq.type === CommentType.externalReference &&
+      commentReq.externalReferenceStorage.type === ExternalReferenceStorageType.so
+    ) {
+      references = [
+        ...references,
+
+        this.getExternalReferenceSoReferences({
+          id: commentReq.externalReferenceId,
+          type: commentReq.externalReferenceStorage.soType,
+        }),
+      ];
+    }
+
     return references;
   }
 
+  private getExternalReferenceSoReferences({ id, type }: { id: string; type: string }) {
```

Comment:
Just to confirm it's not possible to store two osquery visualizations in a single case comment right?

Reply:
I think for the first iteration of the attachment framework we should not support multiple external references. I think the consumer could create a SO that contains a list of all other SO they need to fetch. Also, supporting multiple documents may introduce performance issues for Cases. @patrykkopycinski do you plan to show multiple osquery visualizations? To my understanding, the osquery SO contains all the information needed to render the results, correct?

Reply:
Thanks, @patrykkopycinski! I thought it was an SO 🙂.